.github/actions/build-deps/action.yml

```diff
@@ -0,0 +1,108 @@
+# This action installs and optionally uploads Conan dependencies to a remote
+# repository. The dependencies will only be uploaded if the credentials are
+# provided.
+name: Build Conan dependencies
+
+inputs:
+  build_dir:
+    description: 'The directory where to build.'
+    required: true
+    type: string
+  build_type:
+    description: 'The build type to use.'
+    required: true
+    type: string
+  conan_global_conf:
+    description: 'The contents of the global Conan configuration file.'
+    required: true
+    type: string
+  conan_remote_name:
+    description: 'The name of the Conan remote to use.'
+    required: true
+    type: string
+  conan_remote_url:
+    description: 'The URL of the Conan endpoint to use.'
+    required: true
+    type: string
+  conan_remote_username:
+    description: 'The username for logging into the Conan remote. If not provided, the dependencies will not be uploaded.'
+    required: false
+    type: string
+    default: ''
+  conan_remote_password:
+    description: 'The password for logging into the Conan remote. If not provided, the dependencies will not be uploaded.'
+    required: false
+    type: string
+    default: ''
+  force_build:
+    description: 'Force building of all dependencies.'
+    required: false
+    type: boolean
+    default: false
+  force_upload:
+    description: 'Force uploading of all dependencies.'
+    required: false
+    type: boolean
+    default: false
+  os:
+    description: 'The operating system that is used (Linux, MacOS, or Windows).'
+    required: true
+    type: choice
+    options:
+      - Linux
+      - MacOS
+      - Windows
+
+runs:
+  using: composite
+  steps:
+    # Set up Conan. Note that on MacOS and Windows, which are self-hosted
+    # runners, we should overwrite the global.conf file instead of appending to
+    # it. On Linux runners, where we run Docker images, the file is reset on
+    # each run with important defaults.
+    - name: Install Conan profile
+      shell: bash
+      run: |
+        echo 'Installing configuration and profile.'
+        if [ '${{ inputs.os }}' == 'Linux' ]; then
+          echo '${{ inputs.conan_global_conf }}' >> $(conan config home)/global.conf
```

Comment:
is there some `global.conf` on MacOS or Windows that we want to overwrite (I know there is one on Linux that we want to keep) ?

if not, then perhaps we can make this `>>` unconditional ?